internal/pkg/template/diff/diff.go

```diff
@@ -6,63 +6,83 @@ package diff
 
 import (
 	"fmt"
-	"strconv"
 
 	"gopkg.in/yaml.v3"
 )
 
 // Node represents a segment on a difference between two YAML documents.
-type Node struct {
-	key      string
-	children map[string]*Node // A list of non-empty pointers to the children nodes.
+type Node interface {
```

Comment:
It's a bit odd to have a public interface that has only private methods 🤔 Have we seen something like this in the std lib?
What do you think of an API like this:
```go
// Tree represents a difference tree between two YAML documents.
type Tree struct {
   root diffNode
}

// comparableNode is the interface to represents the difference between two *yaml.Node.
type comparableNode interface {
   key() string
   // your other methods...
}

// node is a concrete implementation of a comparableNode.
type node struct {
   keyValue string
   // your other fields...
} 

func (from From) Parse(to []byte) (*Tree, error)
```